### PR TITLE
Added new RDD API to allow applying readMany even to large (>> 100,000) number of documents

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

This PR adds a new API `com.azure.cosmos.spark.CosmosItemsDataSource.readMany(DataFrame, java.util.Map[String, String], [StructType])` which allows bulk loading documents from Cosmos DB via readMany based on an existing DataFrame (which needs to have a column `_itemIdentity` with pk+id (see UDF `GetCosmosItemIdentityValue`) or `id` (if `/id` is the partition key)

**Scala Sample**
```
// scalastyle:off underscore.import
import scala.collection.JavaConverters._
// scalastyle:on underscore.import
 
val input = spark.read.parquet("dbfs:/data/PK_And_Id.parquet").limit(1000000).repartition(48)
val readCfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
  "spark.cosmos.accountKey" -> cosmosMasterKey,
  "spark.cosmos.database" -> "LongRunTestDB",
  "spark.cosmos.container" -> "TestData"
)
 
val readManyDf = com.azure.cosmos.spark.CosmosItemsDataSource.readMany(input, readCfg.asJava)
println(readManyDf.count())
readManyDf.show()
```

**PySpark Sample**
```
from pyspark.sql import DataFrame

input = spark.read.parquet("dbfs:/data/PK_And_Id.parquet").limit(1000000).repartition(48)
readCfg = {
    "spark.cosmos.accountEndpoint": cosmosEndpoint,
    "spark.cosmos.accountKey": cosmosMasterKey,
    "spark.cosmos.database": "LongRunTestDB",
    "spark.cosmos.container": "TestData"
}

readManyDf =  DataFrame(sc._jvm.com.azure.cosmos.spark.CosmosItemsDataSource.readMany(input._jdf, readCfg), sqlContext)
print(readManyDf.count())
readManyDf.show()
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosItemsDataSource.scala [hunk 1]
```diff
@@ -83,4 +81,35 @@ object CosmosItemsDataSource {
   def version : String = {
     CosmosConstants.currentVersion
   }
+
+  def readMany(df: DataFrame, userConfig: java.util.Map[String, String]): DataFrame = {
+    readMany(df,  userConfig,null)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosItemsDataSource.scala:86]: nit: space

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosReadManyReader.scala [hunk 2]
```diff
@@ -0,0 +1,155 @@
+package com.azure.cosmos.spark
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosReadManyReader.scala:1]: copyright

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosItemsDataSource.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosItemsDataSource.scala:86]: Fixed

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosReadManyReader.scala (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosReadManyReader.scala:1]: Fixed

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks for such a quick turnaround and great solution :)

**Pull request merged**